src/v/raft/state_machine_manager.h

```diff
@@ -126,7 +126,7 @@ class state_machine_manager final {
     }
 
     template<StateMachineIterateFunc Func>
-    void for_each_stm(Func&& func) const {
+    void for_each_stm(const Func& func) const {
```

Comment:
Do we assume that if a function supplied is to be called more than once then the caller should make it immutable and use by-reference capture if they need to accumulate some information between calls?

Reply:
Good point, i will change to non const ref

Reply:
i dropped that change